Sources/AppcuesKit/Data/PushMonitor.swift

```diff
@@ -45,14 +52,18 @@ internal class PushMonitor: PushMonitoring {
         refreshPushStatus()
     }
 
-    private func refreshPushStatus() {
+    func refreshPushStatus(completion: ((UNAuthorizationStatus) -> Void)? = nil) {
```

Comment:
wondering if we need to allow some public interface to this as well, or at least call when the device token is passed in? Thinking that the `pushPrimerEligible` value could change outside of our scope, from the host app doing some permission request on its own, and we'd want to update our value so they don't later trigger a flow that doesn't make sense any longer.

Reply:
Good question, it's a possibility, but not super likely? We can definitely expose something in the public API, but I don't really want a giant list of, "make sue you call these 20 functions in these 20 spots" to make everything integrate seamlessly.